misc/dbt-materialize/tests/adapter/test_deploy.py

```diff
@@ -742,3 +734,14 @@ def test_full_deploy_process(self, project):
         ), "Sink's view ID should be different after deployment"
 
         run_dbt(["run-operation", "deploy_cleanup", "--vars", project_config])
+
+
+def run_with_retry(project, sql, expected_count, retries=5, delay=3, fetch="one"):
+    for i in range(retries):
+        result = project.run_sql(sql, fetch=fetch)
+        if result[0] == expected_count:
+            return result
+        time.sleep(delay)
+    raise AssertionError(
+        f"Expected count to be {expected_count}, but got {result[0]} after {retries} retries"
+    )
```

Comment:
As suggested in #27920 adding a retry for some of those assertions. @def- mentioned that we have something similar in Testdrive, but not 100% sure if we could re-use that here.